configure.ac

```diff
@@ -2946,6 +2946,44 @@ esac
 ],
 [AC_MSG_RESULT(default)])
 
+
+validate_tzpath() {
+    # Check that each element of the tzpath is an absolute path
+    all_paths=$1
+    while test "${all_paths#*:}" != "$1"; do
+        cur_path="${all_paths%%:*}"
+        case "$cur_path" in
+            /*)  ;; # Absolute path
+            *)
+                AC_MSG_ERROR([--with-tzpath must contain only absolute paths])
+                return 1
+                ;;
+        esac
+
+        all_paths="${all_paths#*:}"
+    done
+}
+
+TZPATH="/usr/share/zoneinfo:/usr/lib/zoneinfo:/usr/share/lib/zoneinfo:/etc/zoneinfo"
+AC_MSG_CHECKING(for --with-tzpath)
+AC_ARG_WITH(tzpath,
+            AS_HELP_STRING([--with-tzpath=<list of relative paths separated by pathsep>]
+                           [Select the default time zone search path for zoneinfo.TZPATH]),
+[
+case "$withval" in
+    yes)
+        AC_MSG_ERROR([--with-tzpath requires a value])
+        ;;
+    *)
+        validate_tzpath "$withval"
+        TZPATH="$withval"
+        AC_MSG_RESULT($withval)
+        ;;
+esac
+],
+[AC_MSG_RESULT(default)])
```

Comment:
Add $TZPATH in the message here, so you can see what that default is?

Reply:
Good catch. I originally had it this way, but then I played around with the possibility of using something closer to the previous logic, where if `TZPATH` is not explicitly set, at runtime the module will order the path based on a hard-coded string where the order is determined by which directories exist at import time. In that case, "default" made more sense than any hard-coded string.

I abandoned that approach as YAGNI — if we want to do something like that, we should probably just do it at build time anyway.